documentation/Bundling.md

```diff
@@ -0,0 +1,483 @@
+
+```html
+<script src="./dist/bundle.js"></script>
+```
+
+## Using Parcel
+
+First, you need to install [parcel](https://parceljs.org/) globally:
+
+```
+npm install -g parcel-bundler
+```
+
+Once this is done, you can use parcel by configuring your project in the way that parcel expects.
+
+### Parcel with Javascript
+
+Parcel uses [browserslist](https://github.com/browserslist/browserslist) to configure what polyfills are needed when bundling. Azure SDK libraries generally target the ES2015 version of JavaScript and use some modern features of JavaScript, including [generators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*), so let's edit `package.json` to target the latest version of three popular browsers:
+
+```json
+"browserslist": [
+    "last 1 Chrome version",
+    "last 1 Firefox version",
+    "last 1 Edge version"
+  ],
+```
+
+In order to use Azure SDK libraries inside JS, you need to import code from the package you installed earlier.
+
+To accomplish this, let's create two files, `index.js` and `index.html`:
+
+```js
+// index.js
+const { BlobServiceClient } = require("@azure/storage-blob");
+// Now do something interesting with BlobServiceClient :)
+```
+
+```html
+<!-- index.html -->
+<!DOCTYPE html>
+<html>
+  <body>
+    <script src="./index.js"></script>
+  </body>
+</html>
+```
+
+Now you can invoke parcel on the command-line:
+
+```
+parcel index.html
+```
+
+This will bundle your code and create a local development server for your page at `http://localhost:1234`. Changes you make to `index.js` will automatically get reflected on the dev server.
+
+If you wish to bundle your page without using the local development server, you can do this by passing the `build` command:
+
+```
+parcel build index.html
+```
+
+This will emit a compiled version of `index.html`, as well as any included script files, to the `dist` directory.
+
+### Parcel with TypeScript
+
+Parcel uses [browserslist](https://github.com/browserslist/browserslist) to configure what polyfills are needed when bundling. The Azure SDK uses some modern features of JavaScript, including [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function), so let's edit `package.json` to target the latest version of three popular browsers:
+
+```json
+"browserslist": [
+    "last 1 Chrome version",
+    "last 1 Firefox version",
+    "last 1 Edge version"
+  ],
+```
+
+Next, you need to install [TypeScript](https://typescriptlang.org):
+
+```
+npm install --save-dev typescript
+```
+
+Now let's create a very basic [tsconfig.json](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) file to configure TypeScript:
+
+```json
+{
+  "compilerOptions": {
+    "outDir": "./dist/",
+    "noImplicitAny": true,
+    "strict": true,
+    "module": "es6",
+    "moduleResolution": "node",
+    "target": "es2017"
```

Comment:
minor: does parcel require es2017? Other snippets above are targetting es6

Reply:
ah this is not needed. Was from when I was figuring out the browserslist problem